### ISSUE TITLE
Use <f3> to calculate directory size

### ISSUE THREAD

**Issue by abitrolly** [ranger/ranger#2238]
#### Runtime Environment

- Operating system and version: Fedora 33
- Python version: 3.9.1 (default, Dec  8 2020, 00:00:00) [GCC 10.2.1 20201125 (Red Hat 10.2.1-9)]
- Ranger version/commit:  ranger 1.9.3
- Locale: en_US.UTF-8

#### Current Behavior

Pressing <kbd>F3</kbd> while on dir does nothing.

#### Expected Behavior

Press <kbd>F3</kbd> to calculate dir size. Press kbd>F3</kbd> to get back to previous output.

#### Context

I am trying to see how much my repo takes.

**Comment by toonn** [comment]
Hmm, `<f3>` is mapped to `display_file` by default. Not sure why you expect it to show the directory's size? Maybe you want `dc` or `du`?

**Comment by abitrolly** [comment]
@toonn that's feature request. Right now `<f3>` does nothing for dirs. `dc` is good, but it is hard to discover.

**Comment by toonn** [comment]
Ok, so have `<f3>` use "dc" for dirs but still `display_file` for files? If no one objects or has even better suggestions this'll probably make it in some day : )

**Comment by abitrolly** [comment]
@toonn yes, that will be good. )

**Comment by chu4ng** [comment]
I'm not sure about using Fn keys, because it opposite vim-philoshpy and I think it should be defined by user if he wants it, not included in default config.

**Comment by abitrolly** [comment]
@chu4ng do you propose to remove these?
```
        <f1> help
        <f2> rename_append
        <f3> display_file
        <f4> edit
        <f5> copy
        <f6> cut
        <f7> console mkdir%space
        <f8> console delete
       <f10> exit
```

**Comment by Rom1deTroyes** [comment]
@abitrolly theses shortcuts are the ones from [Midnight Commander](https://midnight-commander.org/) to facilitate the transition for non vim-native users.
![image](https://user-images.githubusercontent.com/8625432/107114510-94f57800-6866-11eb-9433-ef7d335b866a.png)

It's like using arrows key or <kbd>Esc</kbd>+<kbd>Meta</kbd>+<kbd>Alt</kbd>+<kbd>Control</kbd>+<kbd>Shift</kbd>+<kbd>Key</kbd> shortcuts : not in the philosophy, but you can use it if you mind to.

Btw, as the functions are already mapped here, and just enhance the functionality for directory, it may not be harmful to add this ?

**Comment by toonn** [comment]
Yep, adding this doesn't harm the UX imo. It just makes `f3` useful for directories.

**Comment by abitrolly** [comment]
@Rom1deTroyes being a long time Far Manager user I did't like `mc`, because it is slow, but I like `ranger`, because I can modify it, rather than configure. Glad it has more usability features than other Linux file managers. Because I can not carry my dot files to each container I need to access.

**Comment by abitrolly** [comment]
The killer feature of Far Manager that I miss from Linux file managers is using `Alt+<key>` for navigating to matching file. Works like `Ctrl+F` search in Firefox, but already prefilled with the key and scrolled to the first matching result.

**Comment by chu4ng** [comment]
> The killer feature of Far Manager that I miss from Linux file managers is using Alt+<key> for navigating to matching file
Just like `f+pattern` in ranger? (A changed alias find to `scout -aetsf` for filtering files.)

**Comment by abitrolly** [comment]
> Just like f+pattern in ranger? (A changed alias find to scout -aetsf for filtering files.)

Probably that's `f<pattern>`, because `f+` would mean `f` is not released and that's just repeats `f` in that `find:` dialog.

`f<pattern>` opens the file when typing is matched enough. That's absolutely not what I expect from the interface. The `alt+<pattern>` in Far Manager just moves the cursor. And then I could do what I planned to it.

**Comment by chu4ng** [comment]
>just moves the cursor
IMHO, opening files(with pretty rifle settings) it's more comfortable, but who cares.
You can change find-alias to `scout -ts` and may be add `f` flag for filtering. Then it doesn't open file, just moving cursor to file.

**Comment by abitrolly** [comment]
@chu4ng comfortable for opening files only. For managing files and archives, it is not comfortable at all. Speaking of archive management, it is another killer feature of Far Manager - natively navigating inside archives, extracting files from them, copying into, etc.

**Comment by chu4ng** [comment]
>natively navigating inside archives, extracting files from them, copying into, etc.

It's good idea for new module.

**Comment by abitrolly** [comment]
@chu4ng I couldn't figure out what a plugin interface for the `ranger` is. If the API could be automatically translated to Far Manager, it would be easier to just port Far plugins to Linux that rewrite them.

**Comment by chu4ng** [comment]
@abitrolly 
I understand it as copying files(or writing you own) from https://github.com/ranger/ranger/tree/master/examples to $ranger_conf/plugins.
Just like adding classes to $ranger_conf/commands.py, but it looks better. In other words, it just commands defined by user.
But I'm not sure about UI for this functionality.

I don't think what we can translate Far API to ranger.

**Comment by abitrolly** [comment]
@chu4ng UI is the blocking part. I am not sure that `ranger` provide and API at all. At least there is no docs, so it looks just any other internal details.

**Comment by toonn** [comment]
There's no documented API and there's no effort I know of in that direction unfortunately. It was originally added as an afterthought afaict. People still do really cool things with plugins though!

But like @chu4ng said, check out `scout` it probably does what you're looking for and much more, you could also check out `travel` which sounds a little similar to what you're looking for. I wouldn't be surprised if you can mimic the `alt+X` behavior of Far.

Re archive integration, check the wiki maybe, I remember someone doing that with a fuse filesystem it might be on there.

**Comment by abitrolly** [comment]
Thanks for replies. I will take a look.
